pennylane/devices/default_qubit.py

```diff
@@ -1037,7 +1037,7 @@ def eval_jaxpr(
 
         interpreter = DefaultQubitInterpreter(
             num_wires=len(self.wires),
-            shots=self.shots.total_shots,
+            shots=shots or self.shots,
```

Comment:
```suggestion
            shots=Shots(shots).total_shots,
```
Just like with `Device.execute`, we should now *always* take the overridden shots.  

We also need to update the `if self.shots.has_partitioned_shots` error up above.